platform/config/ensureDeepObject.ts

```diff
@@ -1,29 +1,42 @@
 const separator = '.';
 
-export function ensureDeepObject(obj: any) {
+/**
+ * Recursively traverses through the object's properties and expands ones with
+ * dot-separated names into nested objects (eg. { a.b: 'c'} -> { a: { b: 'c' }).
+ * @param obj Object to traverse through.
+ * @returns Same object instance with expanded properties.
+ */
+export function ensureDeepObject(obj: any): any {
```

Comment:
Just noticed that compiler is complaining about implicit return `any` type, so defined it explicitly.